File: packages/core-protocol/src/types/problem-frame.ts
import type { RiskLevel } from "./enums.js";

export interface ProblemFrame {
  id: string;
  run_id: string;
  original_input: string;
  normalized_problem: string;
  constraints: string[];
  success_criteria: string[];
  risk_level: RiskLevel;
  domain: string | null;
  requires_business_decision: boolean;
  requires_research: boolean;
  requires_code: boolean;
  created_at: string;
}

File: services/reasoning-engine/src/mctagents/core/protocol/problem_frame.py
from datetime import datetime

from pydantic import BaseModel, Field

from .enums import RiskLevel


class ProblemFrame(BaseModel):
    id: str = Field(..., description="Unique identifier (UUID)")
    run_id: str = Field(..., description="Reference to the parent run (UUID)")
    original_input: str = Field(..., description="Raw user input")
    normalized_problem: str = Field(..., description="Cleaned problem statement")
    constraints: list[str] = Field(default_factory=list)
    success_criteria: list[str] = Field(default_factory=list)
    risk_level: RiskLevel = RiskLevel.MEDIUM
    domain: str | None = Field(default=None, max_length=64)
    requires_business_decision: bool = False
    requires_research: bool = False
    requires_code: bool = False
    created_at: datetime = Field(default_factory=datetime.now)
